src/tools/perf/perftest_run.c

```diff
@@ -319,6 +319,24 @@ static ucs_status_t run_test_recurs(struct perftest_context *ctx,
     return status;
 }
 
+static void terminate_daemon(struct perftest_context *ctx)
+{
+    ucx_perf_context_t *perf = calloc(1, sizeof(*perf));
+    if (NULL == perf) {
+        ucs_error("Failed to allocate memory for daemon termination");
+        return;
+    }
+
+    perf->params                     = ctx->params.super;
+    perf->params.ucp.is_keep_running = 0;
+    perf->params.command             = UCX_PERF_CMD_AM;
+    perf->params.max_outstanding     = 1;
+
+    ucx_perf_funcs[UCX_PERF_API_UCP].setup(perf);
+    ucx_perf_funcs[UCX_PERF_API_UCP].cleanup(perf);
+    free(perf);
+}
```

Comment:
seems like over complicated to me. maybe just remove keep_running from all places, and kill the daemon from the test script? FIN message would never kill the daemon on itself.

Reply:
let's discuss it

Reply:
per offline discussion, daemon should keep running and close the session with a specific client when getting disconnected (error callback is called on the endpoint)

Reply:
All done, it was quite significant redesign after all.
FIN message is redundant, we can close connection on EP disconnect, so removed FIN completely.
Now daemona are permanent processes (if started manually), which are killed explicitly from test_jenkins.
Also added mechanism to detect stale entries in alias hash by unique ID (which is generated per IB MD, then passed through `uct_ib_md_packed_mkey_t`, then stored in mkey alias).
Process ID does not work for this purpose, because a single process may close MD (and destroy mkey UMR pool), and then reopen it - like happens when we execute test batch.